rest/api_test.go

```diff
@@ -4466,3 +4467,316 @@ func Benchmark_RestApiGetDocPerformanceFullRevCache(b *testing.B) {
+	response = rt.SendAdminRequest(http.MethodPost, "/_profile", "")
+	log.Printf("string(response.BodyBytes()): %v", string(response.BodyBytes()))
+	assertStatus(t, response, http.StatusOK)
+	fi, err = os.Stat(filePath)
+	assert.NoError(t, err, "fetching the file information")
+	assert.True(t, fi.Size() > 0)
+}
+
+func TestHandleHeapProfiling(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	dirPath := fmt.Sprintf("%v/heap-pprof", os.TempDir())
+	assert.NoError(t, os.MkdirAll(dirPath, 0755))
+	defer func() { assert.NoError(t, os.RemoveAll(dirPath)) }()
+
+	// Send a valid request for heap profiling
+	filePath := fmt.Sprintf("%v/heap.pprof", dirPath)
+	reqBodyText := fmt.Sprintf(`{"file":"%v"}`, filePath)
+	response := rt.SendAdminRequest(http.MethodPost, "/_heap", reqBodyText)
+	assertStatus(t, response, http.StatusOK)
+	fi, err := os.Stat(filePath)
+	assert.NoError(t, err, "fetching heap profile file information")
+	assert.True(t, fi.Size() > 0)
+
+	// Send a profile request with invalid json body
+	response = rt.SendAdminRequest(http.MethodPost, "/_heap", "invalid json body")
+	assertStatus(t, response, http.StatusBadRequest)
+	assert.Contains(t, string(response.BodyBytes()), "Invalid JSON")
+
+	// Send profile request with missing JSON 'file' parameter.
+	response = rt.SendAdminRequest(http.MethodPost, "/_heap", "{}")
+	assertStatus(t, response, http.StatusInternalServerError)
+	assert.Contains(t, string(response.BodyBytes()), "no such file or directory")
+}
+
+func TestHandlePprofTrace(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+	// Get and Post requests for pprof trace
+	assert.Panics(t, func() { rt.SendAdminRequest(http.MethodGet, "/_debug/pprof/trace", "") })
+	assert.Panics(t, func() { rt.SendAdminRequest(http.MethodPost, "/_debug/pprof/trace", "") })
+}
+
+func TestHandlePprofs(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	tests := []struct {
+		profileName                 string
+		shortDescription            string
+		inputMethod                 string
+		inputResource               string
+		inputReqBody                string
+		expectedContentDisposition  string
+		expectedContentType         string
+		expectedXContentTypeOptions string
+		expectedStatus              int
+	}{
+		{
+			profileName:                 "goroutine",
+			shortDescription:            "Get request for pprof goroutine",
+			inputMethod:                 http.MethodGet,
+			inputResource:               "/_debug/pprof/goroutine?debug=0&gc=1",
+			inputReqBody:                "",
+			expectedContentDisposition:  `attachment; filename="goroutine"`,
+			expectedContentType:         "application/octet-stream",
+			expectedXContentTypeOptions: "nosniff",
+			expectedStatus:              http.StatusOK},
```

Comment:
There are several values that you expect to be the same for every test (inputReqBody, expectedXContentTypeOptions, etc).  Why are they all being defined in the test struct?

Reply:
This is fixed and the test has been rewritten.